Composer/packages/client/src/utils/publishStatusPollingUpdater.ts

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import httpClient from './httpUtil';
+
+enum CircuitBreakerStateEnum {
+  Closed,
+  Open,
+}
+
+export const ApiStatus = {
+  Publishing: 202,
+  Success: 200,
+  Failed: 500,
+  Unknow: 404,
+};
+
+export class PublishStatusPollingUpdater {
+  private status;
+  private timerId;
+  private targetName;
+  private botProjectId;
+  private pollingInterval = 10000;
+
+  constructor({ botProjectId, targetName }) {
+    this.botProjectId = botProjectId;
+    this.targetName = targetName;
+  }
+
+  async start(onData) {
+    if (!(this.botProjectId && this.targetName)) return;
+    if (this.status === CircuitBreakerStateEnum.Open) return;
+    try {
+      this.status = CircuitBreakerStateEnum.Open;
+
+      const firstResponse = await httpClient.get(`/publish/${this.botProjectId}/status/${this.targetName}`);
+      onData && onData({ botProjectId: this.botProjectId, targetName: this.targetName, apiResponse: firstResponse });
+      this.timerId = window.setInterval(async () => {
+        try {
+          const response = await httpClient.get(`/publish/${this.botProjectId}/status/${this.targetName}`);
+          onData && onData({ botProjectId: this.botProjectId, targetName: this.targetName, apiResponse: response });
+        } catch (err) {
+          console.log(this.timerId);
+          onData && onData({ botProjectId: this.botProjectId, targetName: this.targetName, apiResponse: err.response });
+        }
+      }, this.pollingInterval);
+    } catch (err) {
+      console.log(this.timerId);
+      onData && onData({ botProjectId: this.botProjectId, targetName: this.targetName, apiResponse: err.response });
+    }
+  }
+  stop() {
+    if (this.timerId) {
+      window.clearInterval(this.timerId);
+    }
+    this.timerId = 0;
+    this.status = CircuitBreakerStateEnum.Closed;
+  }
+  restart(onData) {
+    this.stop();
+    this.start(onData);
+  }
+  beEqual(botProjectId, targetName) {
```

Comment:
more descriptive name than beEqual and also space between methods

Reply:
use `isSameUpdater`